src/generators.ts

```diff
@@ -0,0 +1,68 @@
+import axios from 'axios';
+
+/**
+ * Base class for random generators that 
+ * retrieve data from remote sources
+ */
+export class Generator {
+  // url endpoint
+  source: string
+
+  constructor() {
+    this.source = ''
+  }
+
+  // REST API data retrieval from remote source
+  async retrieve(url: string) {
+    return axios.get(url)
+    .then((response) => response.data)
+    .catch(() => {}) 
+  }
+
+  // Subclasses override to produce desired random message
+  async generate() {
+    return ""
+  }
+}
+
+export class InsultGenerator extends Generator {
+  constructor() {
+    super()
+    this.source = 'https://evilinsult.com/generate_insult.php?lang=en&type=json'
```

Comment:
Should these be global constants of some sort? We should have a discussion about if we need to have them as constants in this file or have a dedicated constants file, as other constants have this issue (such as Discord Channel IDs)

Reply:
true, it may be easier to manage in a separate constants file; that's a fair point